src/ImageIO/CGImageMetadata.cs

```diff
@@ -118,15 +118,52 @@ public CGImageMetadata (NSData data)
 			return (result == IntPtr.Zero) ? null : new CGImageMetadataTag (result, true);
 		}
 
+#if NET
+		[DllImport (Constants.ImageIOLibrary)]
+		extern unsafe static void CGImageMetadataEnumerateTagsUsingBlock (/* CGImageMetadataRef __nonnull */ IntPtr metadata,
+			/* CFStringRef __nullable */ IntPtr rootPath, /* CFDictionaryRef __nullable */ IntPtr options,
+			/* __nonnull */ delegate* unmanaged<NativeHandle, NativeHandle, bool> block);
+#else
 		[DllImport (Constants.ImageIOLibrary)]
 		extern static void CGImageMetadataEnumerateTagsUsingBlock (/* CGImageMetadataRef __nonnull */ IntPtr metadata,
 			/* CFStringRef __nullable */ IntPtr rootPath, /* CFDictionaryRef __nullable */ IntPtr options,
 			/* __nonnull */ CGImageMetadataTagBlock block);
+#endif
+
+#if NET
+		static CGImageMetadataTagBlock? callbackBlock;
```

Comment:
You can make this thread local to allow for concurrency at least:

```suggestion
		[ThreadStatic]
		static CGImageMetadataTagBlock? callbackBlock;
```